File: app/src/main/java/com/demo/customview/music/IMusicView.java
package com.demo.customview.music;

import android.content.Context;

import java.util.List;

/**
 * created by 姚明亮
 * Time：2019/8/12 15:51
 */
interface IMusicView {

    /**
     * 初始化
     */
    void init(Context context);

    /**
     * 设置数据
     */

    void setData(List<MusicRow> musicRow);


    /**
     * 指定时间
     */
    void seekTo(int progress, boolean fromUser);

    /**
     * 设置歌词缩放比
     */
    void setMusicScale(float factor);

    /**
     * 重置
     */
    void reset();
}

File: app/src/main/java/com/demo/customview/music/IMusicView.kt
package com.demo.customview.music

import android.content.Context

/**
 * Author by YML, Date on 2021/3/9.
 * PS: Not easy to write code, please indicate.
 */
interface IMusicView {
    /**
     * 初始化
     */
    fun init(context: Context?)

    /**
     * 设置数据
     */
    fun setData(musicRow: ArrayList<MusicRow>)


    /**
     * 指定时间
     */
    fun seekTo(progress: Int, fromUser: Boolean)

    /**
     * 设置歌词缩放比
     */
    fun setMusicScale(factor: Float)

    /**
     * 重置
     */
    fun reset()
}